servers/graphql-kotlin-server/src/main/kotlin/com/expediagroup/graphql/server/execution/GraphQLServer.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2021 Expedia, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.expediagroup.graphql.server.execution
+
+import com.expediagroup.graphql.types.GraphQLResponse
+
+/**
+ * A basic server implementation that parses the incoming request and returns a [GraphQLResponse].
+ * Subscriptions require more server-specific details and should be implemented separately.
+ */
+open class GraphQLServer<Request>(
+    private val requestParser: GraphQLRequestParser<Request>,
+    private val contextFactory: GraphQLContextFactory<*, Request>,
+    private val requestHandler: GraphQLRequestHandler
+) {
+
+    /**
+     * Default execution logic for handling a [Request] and returning a [GraphQLResponse].
+     *
+     * If null is returned, that indicates a problem parsing the request or context.
+     * If the request is valid, a [GraphQLResponse] should always be retuned.
+     * In the case of errors or exceptions, return a response with [GraphQLResponse.errors] populated.
+     * If you need custom logic inside this method you can override this class or choose not to use it.
+     */
+    open suspend fun getResponse(request: Request): GraphQLResponse<*>? {
```

Comment:
wondering whether we should rename this method to `execute` to match naming in request handler `executeRequest` and  `graphql-java` `GraphQL#execute`/`GraphQL#executeAsync`